src/Frame/Frame.cc

```diff
@@ -1489,13 +1503,30 @@ bool Frame::FillSpectralProfileData(std::function<void(CARTA::SpectralProfileDat
                     count(_z_axis) = nz;
                     casacore::Slicer slicer(start, count);
                     const auto N = slicer.length().product();
+#if defined(__APPLE__) && defined(__MACH__) && defined(__MAC_OS_X_VERSION_MIN_REQUIRED)
+#if __MAC_OS_X_VERSION_MIN_REQUIRED < 110000
```

Comment:
As above; I would flatten this into a single conditional.

Reply:
I can't recall if it's possible, but doing some sort of `#define HAS_SHARED_PTR defined(__APPLE__) && defined(__MACH__) && defined(__MAC_OS_X_VERSION_MIN_REQUIRED)` monstrosity might make the actual code more readable.

Reply:
Good point; if we're going to collapse this into one check we can also do this.